### PR TITLE
Consolidate duplicated utility helpers

### PR DESCRIPTION
**Pull request by blckmn**
Consolidates small utility logic that had drifted into copies: adds radToDeg, clamp and METERS_TO_FEET to common.js and replaces the local reimplementations (deg/rad conversion existed in seven places); extracts shared rcChannel.js (900/2100 constants plus clampChannel/channelPercent) for AuxiliaryTab and the adjustments composable; extracts battery.js estimateCellCount for BatteryIcon/BatteryLegend. One behaviour fix: SensorStatus.vue had a stale local copy of have_sensor missing the opticalflow case — it now imports the canonical helper, so the optical flow sensor is recognised there. New vitest coverage for the extracted helpers; full suite green (396 tests).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved consistency across battery cell count display, sensor status indicators, RC channel/slider values, and OSD preview positioning.
  * Refined angle conversion used in calibration and 3D orientation/tilt calculations for more reliable visual behavior.
  * Standardized clamping and input handling to reduce edge-case display issues (including safer battery cell-count estimation for invalid inputs).
* **Tests**
  * Added unit tests for shared utility conversions, clamping behavior, RC channel helpers, and battery cell-count estimation.

### CODE REVIEW

**Review by nerdCopter — APPROVED** [review]
- shallow tested from flashing to config of unsoldered fc. no issues found, but deferring to @blckmn for coderabbitAI nitpick and copliot resolutions,

**Review by haslinghuis — APPROVED** [review]

File: src/components/tabs/pid-tuning/RatesSubTab.vue
Comment (haslinghuis) [inline src/components/tabs/pid-tuning/RatesSubTab.vue]:
```suggestion
```

File: src/js/utils/battery.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const NO_BATTERY_VOLTAGE_MAXIMUM = 1.8;
+
+export function estimateCellCount(voltage, vbatmaxcellvoltage) {
+    if (voltage === 0) {
+        return 1;
+    }
+    return Math.floor(voltage / vbatmaxcellvoltage) + 1;
+}
```
[the remark this answers is not in the document]
Comment (blckmn) [reply src/js/utils/battery.js:8]: Good catch. Switched to Math.ceil with a guard that returns 1 when either input is zero, negative or non-finite, so the 0-init vbatmaxcellvoltage case can't produce Infinity. Added tests for the exact-boundary and unset-max cases.

File: src/components/tabs/pid-tuning/RatesSubTab.vue
Comment (blckmn) [reply src/components/tabs/pid-tuning/RatesSubTab.vue]: Removed. It was unused and referenced API_VERSION_1_45 which wasn't even imported.
